src/main/scala/viper/gobra/frontend/info/implementation/typing/ImportTyping.scala

```diff
@@ -7,15 +7,29 @@
 package viper.gobra.frontend.info.implementation.typing
 
 import org.bitbucket.inkytonik.kiama.util.Messaging.{message, noMessages}
-import viper.gobra.ast.frontend.{PExplicitQualifiedImport, PImplicitQualifiedImport, PImport, PUnqualifiedImport}
+import viper.gobra.ast.frontend.{PExplicitQualifiedImport, PImplicitQualifiedImport, PImport, PNode, PUnqualifiedImport}
+import viper.gobra.frontend.PackageResolver.RegularImport
 import viper.gobra.frontend.info.implementation.TypeInfoImpl
 
 trait ImportTyping extends BaseTyping { this: TypeInfoImpl =>
 
-  lazy val wellDefImport: WellDefinedness[PImport] = createWellDef {
-    case _: PExplicitQualifiedImport => noMessages
-    case _: PUnqualifiedImport => noMessages
-    // this case should never occur as these nodes should get converted in the parse postprocessing step
-    case n: PImplicitQualifiedImport => message(n, s"Explicit qualifier could not be derived")
+  lazy val wellDefImport: WellDefinedness[PImport] = createWellDef { imp =>
+    forceNonLazyImport(imp.importPath, imp)
```

Comment:
I think that this question can be interpreted in two ways:
1. **can we ignore all packages `X` whose files contain only imports without import preconditions?** No, because we need to establish if imports of package `X` satisfy the init postcondition when conjoined.
2. (most likely) **can we ignore all packages `X` for which all `import X` in other packages have `true` as precondition?** - I am not sure. I don't think so, because even in that case, a package can have import preconditions which should be taken into consideration.

I do think that it is possible to avoid desugaring (but not type-checking) the whole package tree, but I do think that would rely on a more fine-grained analysis over which methods can be skipped and which methods cannot - I do not think that a course-grained analysis at the package level would be very effective. I would leave this to a future PR, as previous attempts at selective importing have shown that this is not quite so trivial.